### ISSUE TITLE
Dialyzer warnings

### ISSUE THREAD

**Issue by tuncer** [triqng/triq#32]
```erlang
==> triq (dialyze)
DEBUG: dialyze opts:
[{analysis_type,succ_typings},
 {check_plt,false},
 {init_plt,"/builds/triqng/triq/.rebar/triq_17.5.6.9_plt"},
 {files_rec,["ebin"]},
 {warnings,[]}]
Unknown function compile:forms/2
Unknown function crypto:strong_rand_bytes/1
Unknown function eunit:test/1
Unknown function triq_rnd:seed/0
Unknown function triq_rnd:seed/1
Unknown function triq_rnd:uniform/0
Unknown function triq_rnd:uniform/1
src/triq_dom.erl:239: Cons will produce an improper list since its 2nd argument is {#'@'{kind::#choose{min::integer(),max::integer()},pick::fun((#'@'{kind::{_,_,_},pick::fun((_,_) -> any()),shrink::fun((_,_) -> any()),empty_ok::boolean()},integer()) -> {{_,_,_,_,_},number()}),shrink::fun((#'@'{kind::{_,_,_},pick::fun((_,_) -> any()),shrink::fun((_,_) -> any()),empty_ok::boolean()},integer()) -> {{_,_,_,_,_},integer()}),empty_ok::'true'}}
src/triq_dom.erl:279: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:333: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:351: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:374: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:445: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:446: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:454: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:558: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:894: The pattern 0 can never match the type pos_integer()
src/triq_dom.erl:920: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:983: Function bindshrink/2 has no local return
src/triq_dom.erl:986: The call triq_dom:domain('letshrink',fun((_,_) -> none()),'undefined') will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> #'@'{kind::atom() | tuple(),pick::fun((_,integer()) -> {_,_}),shrink::fun((_,_) -> {_,_}),empty_ok::'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_dom.erl:987: The created fun has no local return
src/triq_dom.erl:989: The variable _ can never match since previous clauses completely covered the type 'false'
src/triq_dom.erl:996: Function bindshrink2/5 has no local return
src/triq_dom.erl:997: The call triq_dom:domain('letshrink2','undefined',fun((_,_) -> {_,_})) will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> #'@'{kind::atom() | tuple(),pick::fun((_,integer()) -> {_,_}),shrink::fun((_,_) -> {_,_}),empty_ok::'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_dom.erl:1077: Cons will produce an improper list since its 2nd argument is #'@'{kind::#sized{},pick::fun(({_,_,_,_,_},pos_integer()) -> {_,_}),shrink::fun((_,_) -> none()),empty_ok::'true'}
src/triq_dom.erl:1100: Function frequency/1 has no local return
src/triq_dom.erl:1107: The call triq_dom:domain('frequency',fun((_,_) -> {_,_}),'undefined') will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> #'@'{kind::atom() | tuple(),pick::fun((_,integer()) -> {_,_}),shrink::fun((_,_) -> {_,_}),empty_ok::'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_dom.erl:1204: Function shrink_without_duplicates/1 has no local return
src/triq_dom.erl:1205: The call triq_dom:domain('shrink_without_duplicates1',fun((_,_) -> none()),'undefined') will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> #'@'{kind::atom() | tuple(),pick::fun((_,integer()) -> {_,_}),shrink::fun((_,_) -> {_,_}),empty_ok::'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_dom.erl:1206: The created fun has no local return
src/triq_dom.erl:1213: Function shrink_without_duplicates/2 has no local return
src/triq_dom.erl:1214: The call triq_dom:domain('shrink_without_duplicates2','undefined',fun((_,_) -> any())) will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> #'@'{kind::atom() | tuple(),pick::fun((_,integer()) -> {_,_}),shrink::fun((_,_) -> {_,_}),empty_ok::'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_dom.erl:1457: Function unicode_binary/0 has no local return
src/triq_dom.erl:1458: The call triq_dom:unicode_binary('any','utf8') breaks the contract (Size,Encoding) -> domrec(binary()) when is_subtype(Size,non_neg_integer() | 'undefined'), is_subtype(Encoding,unicode:encoding())
src/triq_dom.erl:1469: The call triq_dom:unicode_binary('any',Encoding::any()) breaks the contract (Size,Encoding) -> domrec(binary()) when is_subtype(Size,non_neg_integer() | 'undefined'), is_subtype(Encoding,unicode:encoding())
src/triq_dom.erl:1478: The call triq_dom:unicode_binary('any',Encoding::any()) breaks the contract (Size,Encoding) -> domrec(binary()) when is_subtype(Size,non_neg_integer() | 'undefined'), is_subtype(Encoding,unicode:encoding())
src/triq_dom.erl:1525: The created fun has no local return
src/triq_dom.erl:1538: The created fun has no local return
src/triq_dom.erl:1538: The call triq_dom:unicode_characters(Encoding::non_neg_integer()) breaks the contract (Encoding) -> domrec(uchars()) when is_subtype(Encoding,unicode:encoding())
src/triq_fsm.erl:33: Function commands/1 has no local return
src/triq_fsm.erl:36: Function commands/2 has no local return
src/triq_statem.erl:59: Function commands/1 has no local return
src/triq_statem.erl:60: The call triq_dom:domain('commands',fun((_,_) -> no_return()),'undefined') will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> {'@',atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_}),'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_statem.erl:69: Function commands/2 has no local return
src/triq_statem.erl:70: The call triq_dom:domain('commands',fun((_,_) -> no_return()),'undefined') will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> {'@',atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_}),'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_statem.erl:110: Function shrinkable_commands/3 has no local return
src/triq_statem.erl:111: The call triq_dom:domain('shrinking_commands','undefined',fun((_,_) -> {_,_})) will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> {'@',atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_}),'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_statem.erl:249: Function prop_statem/1 has no local return
src/triq_statem.erl:250: The created fun has no local return
src/triq_statem.erl:266: The pattern <RemIdx, Tup> can never match since previous clauses completely covered the type <_,[any()]>
src/triq_statem_fsm.erl:46: Function commands/1 has no local return
src/triq_statem_fsm.erl:47: The call triq_dom:domain('commands',fun((_,_) -> no_return()),'undefined') will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> {'@',atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_}),'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_statem_fsm.erl:56: Function commands/2 has no local return
src/triq_statem_fsm.erl:57: The call triq_dom:domain('commands',fun((_,_) -> no_return()),'undefined') will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> {'@',atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_}),'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_statem_fsm.erl:97: Function shrinkable_commands/3 has no local return
src/triq_statem_fsm.erl:98: The call triq_dom:domain('shrinking_commands','undefined',fun((_,_) -> {_,_})) will never return since the success typing is (atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_})) -> {'@',atom() | tuple(),fun((_,integer()) -> {_,_}),fun((_,_) -> {_,_}),'true'} and the contract is (Name::atom(),PickFun::pick_fun(T),ShrinkFun::shrink_fun(T)) -> domain(T)
src/triq_statem_fsm.erl:236: Function prop_statem/1 has no local return
src/triq_statem_fsm.erl:237: The created fun has no local return
src/triq_statem_fsm.erl:253: The pattern <RemIdx, Tup> can never match since previous clauses completely covered the type <_,[any()]>
```

**Comment by tuncer** [comment]
Start addressing the issues: #35

**Comment by tuncer** [comment]
Addressed the rest: #50.
